flexmeasures/data/models/planning/storage.py

```diff
@@ -188,16 +199,262 @@ def _prepare(self, skip_validation: bool = False) -> tuple:  # noqa: C901
+                fill_sides=True,
+            )
+
+            # Set up commitments DataFrame
+            commitment = Commitment(
+                name="production peak",
+                quantity=-ems_peak_production,  # production is negative quantity
+                # negative price because peaking in the downwards (production) direction is penalized
+                downwards_deviation_price=-ems_peak_production_price,
+                _type="any",
+                index=index,
+            )
+            commitments.append(commitment)
+
+        # Set up capacity breach commitments and EMS capacity constraints
+        ems_consumption_breach_price = self.flex_context.get(
+            "ems_consumption_breach_price"
+        )
+
+        ems_production_breach_price = self.flex_context.get(
+            "ems_production_breach_price"
+        )
+
+        ems_constraints = initialize_df(
+            StorageScheduler.COLUMNS, start, end, resolution
+        )
+        if ems_consumption_breach_price is not None:
+
+            # Convert to Series
+            ems_consumption_breach_price = get_continuous_series_sensor_or_quantity(
+                variable_quantity=ems_consumption_breach_price,
+                actuator=sensor,
+                unit=(
+                    ems_consumption_breach_price.unit
+                    if isinstance(ems_consumption_breach_price, Sensor)
+                    else (
+                        ems_consumption_breach_price[0]["value"].units
+                        if isinstance(ems_consumption_breach_price, list)
+                        else str(ems_consumption_breach_price.units)
+                    )
+                ),
+                query_window=(start, end),
+                resolution=resolution,
+                beliefs_before=belief_time,
+                fallback_attribute="ems-consumption-breach-price",
+                fill_sides=True,
+            )
+
+            # Set up commitments DataFrame to penalize any breach
+            commitment = Commitment(
+                name="any consumption breach",
+                quantity=ems_consumption_capacity,
+                # positive price because breaching in the upwards (consumption) direction is penalized
+                upwards_deviation_price=ems_consumption_breach_price,
+                _type="any",
+                index=index,
+            )
+            commitments.append(commitment)
+
+            # Set up commitments DataFrame to penalize each breach
+            commitment = Commitment(
+                name="all consumption breaches",
+                quantity=ems_consumption_capacity,
+                # positive price because breaching in the upwards (consumption) direction is penalized
+                upwards_deviation_price=ems_consumption_breach_price,
+                index=index,
+            )
+            commitments.append(commitment)
+
+            # Take the physical capacity as a hard constraint
```

Comment:
Is this consequence of adding value(s) for `ems_consumption_breach_price` clear to the user (in the docs)?

Reply:
I'm adding notes to the `flex-context` section of our docs, to both the capacity field and the price field.